temporian/core/data/sampling.py

```diff
@@ -15,54 +15,90 @@
 """A sampling."""
 
 from __future__ import annotations
+from enum import Enum
 from typing import Dict, List, NamedTuple, Optional, Tuple, TYPE_CHECKING, Union
 
-from temporian.core.data import dtype as dtype_lib
+from temporian.core.data.dtype import DType
 
 if TYPE_CHECKING:
     from temporian.core.operators.base import Operator
 
 
-IndexDtypes = Union[dtype_lib.INT32, dtype_lib.INT64, dtype_lib.STRING]
+class IndexDTypes(Enum):
+    INT32 = DType.INT32.value
+    INT64 = DType.INT64.value
+    STRING = DType.STRING.value
 
 
 class IndexLevel(NamedTuple):
     name: str
-    dtype: IndexDtypes
+    dtype: IndexDTypes
+
+
+class Index:
+    def __init__(
+        self,
+        levels: Union[Tuple[str, IndexDTypes], List[Tuple[str, IndexDTypes]]],
+    ) -> None:
+        if isinstance(levels, tuple):
```

Comment:
nit: use `isinstance(..., Tuple)` instead of `tuple` for consistency with constructor?